core/src/banking_stage.rs

```diff
@@ -75,33 +76,131 @@ impl BankingStage {
 
     fn forward_unprocessed_packets(
         socket: &std::net::UdpSocket,
-        tpu: &std::net::SocketAddr,
-        unprocessed_packets: UnprocessedPackets,
+        forwarder: &std::net::SocketAddr,
+        unprocessed_packets: &[(SharedPackets, usize)],
     ) -> std::io::Result<()> {
+        let mut blob = Blob::default();
         for (packets, start_index) in unprocessed_packets {
             let packets = packets.read().unwrap();
-            for packet in packets.packets.iter().skip(start_index) {
-                socket.send_to(&packet.data[..packet.meta.size], tpu)?;
+            let mut current_index = *start_index;
+            while current_index < packets.packets.len() {
+                current_index += blob.store_packets(&packets.packets[current_index..]) as usize;
+                if current_index < packets.packets.len() {
+                    // Blob is full, send it
+                    socket.send_to(&blob.data[..blob.meta.size], forwarder)?;
+                    blob = Blob::default();
+                } else {
+                    break;
+                }
             }
         }
+
+        if blob.size() > 0 {
+            socket.send_to(&blob.data[..blob.meta.size], forwarder)?;
+        }
+
         Ok(())
     }
 
+    fn forward_buffered_packets(
+        socket: &std::net::UdpSocket,
+        poh_recorder: &Arc<Mutex<PohRecorder>>,
+        cluster_info: &Arc<RwLock<ClusterInfo>>,
+        buffered_packets: &[(SharedPackets, usize)],
+    ) -> bool {
+        let (leader, my_id) = {
+            let rcluster_info = cluster_info.read().unwrap();
+            let leader_id = if let Some(leader) = rcluster_info.leader_data() {
```

Comment:
This code uses 3 different names for the same thing, none of which describes what it is, the leader's ContactInfo.  Also, this code is unnecessarily verbose. How about:

```rust
let leader_info = rcluster_info.leader_data().map(Clone::clone);
```

But why clone the whole contact info?  Could it be:

```rust
let leader_id = rcluster_info.leader_data().map(|x| x.id);
```